### ISSUE TITLE
WebDriverException when no proxy config defined

### ISSUE THREAD

**Issue by priidikvaikla** [Codeception/Codeception#2651]
If no proxy capabilities are defined in acceptance.suite.yml I get 
```
[Facebook\WebDriver\Exception\WebDriverException]
JSON decoding of remote response failed.
Error code: 4
The response: 'TypeError - null is not an object (evaluating 'proxyCapability["proxyType"]')'
```

https://github.com/Codeception/Codeception/blob/2.1/src/Codeception/Module/WebDriver.php#L195 I get null from $this->getProxy() and $this->capabilities contents are set
```
array(3) {
  ["unexpectedAlertBehaviour"]=>
  string(6) "accept"
  ["browserName"]=>
  string(9) "phantomjs"
  ["proxy"]=>
  NULL
}
```

Is there somekind of mistake or should the documentation be updated?

**Comment by Naktibalda** [comment]
It is a bug.

**Comment by tivnet** [comment]
To define a "no proxy" in the yml:

```
    config:
        WebDriver:
            ...
            http_proxy: direct
```

**Comment by ashnazg** [comment]
This bug manifests itself on the Chrome driver in this manner:
`Facebook\WebDriver\Exception\UnknownServerException: unknown error: cannot parse capability: proxy
from unknown error: must be a dictionary`

This was a showstopper for the Chrome driver, as it would not even reach the point of opening the browser.  I tested it against driver versions v2.20 all the way back to 2.14.

I can confirm that @tivnet 's suggestion of `http_proxy: direct` is a good workaround against the 2.1.5 release.

I can confirm that this fix (as tested in the current 2.1.x-dev at a687fba) does solve the issue, without needing the extra config as a workaround.

Is a 2.1.6 expected soon, to get this fix released to us Chrome testers?

**Comment by jepster** [comment]
If I specify 

http_proxy: direct

I'm getting only empty pages with webdriver (PhantomJS). The error is then:

[ModuleException] WebDriver: Current url is blank, no page was opened

Tested different urls. I'm in a Vagrant box. Does I "have" to specify any proxy?

**Comment by tivnet** [comment]
I wrote about "http_proxy", but later I started getting all kind of weird errors, so finally I downgraded codeception and live happily ever since.

**Comment by ashnazg** [comment]
@tivnet , I had another big issue on 2.1.5 too ... see #2684.

**Comment by jepster** [comment]
@tivnet: Which version is working for you? Do you know where I can get the phar archive for this version? If not - no problem, I would download it via composer. But prefering phar at the moment. :)

**Comment by ashnazg** [comment]
@jepster :  right now, the only 2.1.x that's been a problem for me is 2.1.5.  Thus, in my own composer.json, I have:
```
"codeception/codeception": "~2.1 !=2.1.5",
```

All versions of PHARs are listed at http://codeception.com/builds.

I'm hoping that #2684 will get merged and a 2.1.6 release goes out soon :+1:

**Comment by tivnet** [comment]
@jepster I downgraded with composer

"codeception/codeception": "2.1.4"

Remember:
"minimum-stability": "stable",

is a must. WebDriver won't work otherwise.

**Comment by jepster** [comment]
Thanks! Installed Codeception globally via composer by the following entry:

"codeception/codeception": "~2.1 !=2.1.5"

I can confirm that it works now.

**Comment by freinn** [comment]
You saved my life bros. Thank you.

**Comment by agarzon** [comment]
I can confirm this issue. My solution was to downgrade to 2.1.4

**Comment by FelikZ** [comment]
2.1.6 - still broken

**Comment by Naktibalda** [comment]
Fixed by #2660
